model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1338,3 +1359,44 @@ message ExecutableStagePayload {
     string local_name = 2;
   }
 }
+
+// Window information in WindowedValue
+message WindowInfo {
+  // timestamp in milliseconds
+  int64 timestamp = 1;
+  repeated BoundedWindow bounded_windows = 2;
+  PaneInfo pane_info = 3;
+}
+
+// Represents window information assigned to data elements
+message BoundedWindow {
```

Comment:
This limits the set of window types and the number of windows that can be specified. I think we should go with a model where we encode a "dummy element" which is replaced by the actual element when decoding. For example, the payload would be:
```
message ParameterizedWindowedValuePayload {
  // A specification of a windowed value coder where the element coder is always "beam:coder:bytes:v1".
  string coder_id;
  // Contains an encoded windowed value with an empty byte[] element. This elements windowing
  // information should be copied to all elements that are decoded with this coder.
  bytes value;
}
```

This allows us to use the full range of window encodings and the already defined pane/timestamp encodings.